runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java

```diff
@@ -1082,11 +1083,106 @@ private void declareConfigurationProperties(ExtensionDeclarer extensionDeclarer)
 
   private void declareNotifications(ExtensionDeclarer extensionDeclarer) {
     // TODO MULE-17778: Complete this declaration
-    extensionDeclarer.withConstruct("notifications")
+    ConstructDeclarer notificationsConstructDeclarer = extensionDeclarer.withConstruct("notifications")
         .allowingTopLevelDefinition()
         .withStereotype(newStereotype("NOTIFICATIONS", "MULE").withParent(APP_CONFIG).build())
         .describedAs("Registers listeners for notifications and associates interfaces with particular events.")
         .withDeprecation(new ImmutableDeprecationModel("Only meant to be used for backwards compatibility.", "4.0", "5.0"));
+
+    notificationsConstructDeclarer.onDefaultParameterGroup()
+        .withOptionalParameter("dynamic")
+        .ofType(BOOLEAN_TYPE)
+        .withExpressionSupport(NOT_SUPPORTED)
+        .defaultingTo(false)
+        .describedAs("If the notification manager is dynamic, listeners can be registered dynamically at runtime via the "
+            + "MuleContext, and the configured notification can be changed. Otherwise, some parts of Mule will cache "
+            + "notification configuration for efficiency and will not generate events for newly enabled notifications or "
+            + "listeners. The default value is false.");
+
+    // TODO: min-max occurs
+    declareEnableNotification(notificationsConstructDeclarer.withOptionalComponent("notification"));
+
+    // TODO: min-max occurs
+    declareDisableNotification(notificationsConstructDeclarer.withOptionalComponent("disable-notification"));
+
+    // TODO: min-max occurs
+    declareNotificationListener(notificationsConstructDeclarer.withOptionalComponent("notification-listener"));
+  }
+
+  private void declareEnableNotification(NestedComponentDeclarer enableNotificationDeclarer) {
+    enableNotificationDeclarer
+        .describedAs("Registers listeners for notifications and associates interfaces with particular events");
+
+    enableNotificationDeclarer.onDefaultParameterGroup()
+        .withOptionalParameter("event-class")
+        .ofType(STRING_TYPE)
+        .withExpressionSupport(NOT_SUPPORTED)
+        .describedAs("The class associated with a notification event that will be delivered to the interface.\n"
+            + "This can be used instead of the 'event' attribute to specify a custom class.");
+
+    enableNotificationDeclarer.onDefaultParameterGroup()
+        .withOptionalParameter("event")
+        .ofType(STRING_TYPE)
+        .withExpressionSupport(NOT_SUPPORTED)
+        .describedAs("The notification event to deliver.");
+
+    enableNotificationDeclarer.onDefaultParameterGroup()
+        .withOptionalParameter("interface-class")
```

Comment:
camelCase

Reply:
It would break backwards:
```
<notification
                event-class="org.mule.test.config.spring.parsers.specific.ServerNotificationManagerTestCase$TestEvent"
                interface-class="org.mule.test.config.spring.parsers.specific.ServerNotificationManagerTestCase$TestInterface"/>
```